python/nav/portadmin/management.py

```diff
@@ -38,15 +37,11 @@ def get_instance(cls, netbox: manage.Netbox, **kwargs) -> ManagementHandler:
         if not netbox.type:
             raise NoNetboxTypeError()
 
-        vendor_id = netbox.type.get_enterprise_id()
-        if (
-            vendor_id == VENDOR_ID_CISCOSYSTEMS
-            and Cisco.OTHER_ENTERPRISES.is_a_prefix_of(netbox.type.sysobjectid)
-        ):
-            # Cisco SM and SMB products are not supported by the Cisco handler:
-            vendor_id = None
-        handler = VENDOR_MAP.get(vendor_id, SNMPHandler)
-        return handler(netbox, **kwargs)
+        for handler_class in SUPPORTED_HANDLERS:
+            if handler_class.can_handle(netbox):
+                break
+
+        return handler_class(netbox, **kwargs)
```

Comment:
So if no handlers return `True`, then `SNMPHandler` is used as `handler_class` as its the last value in `SUPPORTED_HANDLERS`. Is it possible for SNMPHandler.can_handle to return `False`? It thats possible then it would feel weird for a handler that returned `False `to be assigned as the handler class. 

I preferred how the previous implementation made it clear that `SNMPHandler` is the default value if no others match, declaring that by putting it at the end of a tuple is a bit too indirect I think

Reply:
I agree with that, we should set it as default and then only override it if another handler_class returns True for `can_handle`

Reply:
Yeah, I guess that bit is a bit smelly :laughing: New suggestion pushed.